contracts/kernel/Kernel.sol

```diff
@@ -167,10 +177,11 @@ contract Kernel is IKernel, KernelStorage, Initializable, IsContract, AppProxyFa
     * @param _where Address of the app
     * @param _what Identifier for a group of actions in app
     * @param _how Extra data for ACL auth
-    * @return boolean indicating whether the ACL allows the role or not
+    * @return Boolean indicating whether the ACL allows the role or not.
+    *         Always returns false if the kernel hasn't been initialized yet.
     */
     function hasPermission(address _who, address _where, bytes32 _what, bytes _how) public view returns (bool) {
-        return acl().hasPermission(_who, _where, _what, _how);
+        return hasInitialized() && acl().hasPermission(_who, _where, _what, _how);
```

Comment:
I'm hesitant about adding an extra `SLOAD` for a function that will be called as often as `kernel.hasPermission(...)`. In all 'write functions' we just revert if they are not initialized.

A cheaper way to do it is to check if the ACL address is 0 and if so return false.

Reply:
See c128077